modules/ceph/src/main/java/org/testcontainers/containers/CephContainer.java

```diff
@@ -0,0 +1,133 @@
+package org.testcontainers.containers;
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.client.builder.AwsClientBuilder;
+import org.testcontainers.containers.wait.LogMessageWaitStrategy;
+
+import java.time.Duration;
+
+public class CephContainer<SELF extends CephContainer<SELF>> extends GenericContainer<SELF> {
+
+    private static final String IMAGE = "ceph/demo";
+
+    private String awsAccessKey = "ceph";
+
+    private String awsSecretKey = "ceph";
+
+    private String bucketName = "CEPH";
+
+    private String rgwName = "localhost";
+
+    private String demoUid = "ceph";
+
+    private NetworkAutoDetectMode networkAutoDetectMode = NetworkAutoDetectMode.IPV4_ONLY;
+
+    private Duration startupTimeout = Duration.ofSeconds(20);
+
+    public CephContainer() {
+        super(IMAGE + ":latest");
+    }
+
+    public CephContainer(String dockerImageName) {
+        super(dockerImageName);
+    }
+
+    @Override
+    protected void configure() {
+        withEnv("RGW_NAME", rgwName);
+        withEnv("NETWORK_AUTO_DETECT", networkAutoDetectMode.value);
+        withEnv("CEPH_DEMO_UID", demoUid);
+        withEnv("CEPH_DEMO_ACCESS_KEY", awsAccessKey);
+        withEnv("CEPH_DEMO_SECRET_KEY", awsSecretKey);
+        withEnv("CEPH_DEMO_BUCKET", bucketName);
+        withExposedPorts(6789, 6800, 6801, 6802, 6803, 6804, 6805, 80, 5000);
+        waitingFor(
+                new LogMessageWaitStrategy()
+                        .withRegEx(".*\\/entrypoint.sh: SUCCESS\n")
```

Comment:
Ceph has a built-in health indicator, please use Http wait strategy with `/api/v0.1/health` on port 5000

Reply:
it's not obvious
```java
protected Integer getLivenessCheckPort() {
        if (exposedPorts.size() > 0) {
            return getMappedPort(exposedPorts.get(0));
        } else if (portBindings.size() > 0) {
            return Integer.valueOf(PortBinding.parse(portBindings.get(0)).getBinding().getHostPortSpec());
        } else {
            return null;
        }
    }
```